redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseDispatchers.kt

```diff
@@ -15,21 +15,24 @@
  */
 package app.cash.redwood.treehouse
 
+import kotlin.coroutines.CoroutineContext
 import kotlin.native.ObjCName
-import kotlinx.coroutines.CoroutineDispatcher
 
 /**
  * One of the trickiest things Treehouse needs to do is balance its two dispatchers:
  *
- *  * [ui] is the [CoroutineDispatcher] that runs on the platform's UI thread.
- *  * [zipline] is where downloaded code executes.
+ *  * [ui] executes dispatched tasks on the platform's UI thread.
+ *  * [zipline] executes dispatched tasks on the thread where downloaded code executes.
  *
  * This class makes it easier to specify invariants on which dispatcher is expected for which work.
  */
 @ObjCName("TreehouseDispatchers", exact = true)
 public interface TreehouseDispatchers {
-  public val ui: CoroutineDispatcher
-  public val zipline: CoroutineDispatcher
+  /** Must contain a non-null [kotlinx.coroutines.CoroutineDispatcher]. */
+  public val ui: CoroutineContext
+
+  /** Must contain a non-null [kotlinx.coroutines.CoroutineDispatcher]. */
+  public val zipline: CoroutineContext
```

Comment:
These comments feel weird as we're saying that implementations should be a `CoroutineDispatcher`, when the type is a `CoroutineContext`. Is it really a requirement that a `CoroutineDispatcher` instance be passed?

Reply:
Ooooh good call out. Lemme fix those. (I had temporarily made that change, and then rolled it back.)

Reply:
Actually, as of this PR I haven’t rolled it back, but that happens in https://github.com/cashapp/redwood/pull/1655

I was trying to find a good way to include a CoroutineExceptionHandler, but I found it was easier to put that in the CoroutineScope instead of the CoroutineDispatcher. These CoroutineContext APIs are very dynamic so I tried it both ways and I like the other way better!